newapi/ooniapi/auth.py

```diff
@@ -0,0 +1,406 @@
+        UPDATE SET role = EXCLUDED.role
+    """
+    # TODO: when role is changed enforce token expunge
+    query_params = dict(account_id=account_id, role=role)
+    q = current_app.db_session.execute(query, query_params).rowcount
+    current_app.db_session.commit()
+    # TODO: return update/insert count
+    return q
+
+
+@auth_blueprint.route("/api/v1/set_account_role", methods=["POST"])
+@role_required("admin")
+def set_account_role():
+    """Set a role to a given account identified by an email address
+    ---
+    parameters:
+      - in: body
+        name: email address and role
+        description: data as HTML form or JSON
+        required: true
+        schema:
+          type: object
+          properties:
+            email_address:
+              type: string
+            role:
+              type: string
+    responses:
+      200:
+        description: Confirmation
+    """
+    log = current_app.logger
+    req = request.json if request.is_json else request.form
+    role = req.get("role", "").strip().lower()
+    email_address = req.get("email_address", "").strip().lower()
+    if EMAIL_RE.fullmatch(email_address) is None:
+        return jerror("Invalid email address")
+    if role not in ["user", "admin"]:
+        return jerror("Invalid role")
+
+    r = _set_account_role(email_address, role)
+    log.info(f"Role set {r}")
+    return ""
+
+
+def _delete_account_data(email_address: str) -> None:
+    account_id = hash_email_address(email_address)
+    query = "DELETE FROM accounts WHERE account_id = :account_id"
+    query_params = dict(account_id=account_id)
+    q = current_app.db_session.execute(query, query_params).rowcount
+    current_app.db_session.commit()
+    # TODO return status
+    return q
+
+
+def _get_account_role(account_id: str) -> Optional[str]:
+    """Get account role from database, or None"""
+    query = "SELECT role FROM accounts WHERE account_id = :account_id"
+    query_params = dict(account_id=account_id)
+    q = current_app.db_session.execute(query, query_params)
+    r = q.fetchone()
+    if r:
+        return r[0]
+
+
+@auth_blueprint.route("/api/v1/get_account_role/<email_address>")
+@role_required("admin")
+def get_account_role(email_address):
+    """Get account role. Return an empty string if the account is not found.
+    ---
+    parameters:
+      - name: email_address
+        in: path
+        required: true
+        type: string
+    """
+    log = current_app.logger
+    email_address = email_address.strip().lower()
+    if EMAIL_RE.fullmatch(email_address) is None:
+        return jerror("Invalid email address")
+    account_id = hash_email_address(email_address)
+    role = _get_account_role(account_id)
+    if role is None:
+        log.info(f"Getting account {account_id} role: not found")
+        return ""
+
+    log.info(f"Getting account {account_id} role: {role}")
+    return role
+
+
+@auth_blueprint.route("/api/v1/set_session_expunge", methods=["POST"])
```

Comment:
`<bikeshed>`
Maybe this API endpoint can be called `/api/v1/refresh_session`?
`</bikeshed>`

Reply:
The refreshing is done by the browser of the user. This entry point is used by admins to flag sessions to be expunged and force a refresh if/when the user comes back.